cpp/src/util/main/UnitTests.cpp

```diff
@@ -125,6 +125,343 @@ TEST(eigen_util, sort_columns) {
+  }
+}
+
+TEST(eigen_util, cwiseMax) {
+  constexpr int M = 2;
+  constexpr int N = 4;
+  using Tensor = eigen_util::FTensor<Eigen::Sizes<M, N>>;
+  using Array = Eigen::Array<float, M, N, Eigen::RowMajor>;
+
+  Tensor tensor;
+  tensor.setValues({{0, 1, 2, 3}, {4, 5, 6, 7}});
+  Array expected = {{2, 2, 2, 3}, {4, 5, 6, 7}};
+
+
+  auto result = eigen_util::cwiseMax<Tensor>(tensor, 2);
+
+  for (int i = 0; i < M; ++i) {
+    for (int j = 0; j < N; ++j) {
+      EXPECT_EQ(result(i, j), expected(i, j));
+    }
+  }
+}
+
+TEST(eigen_util, reinterpret_as_tensor) {
+  constexpr int N = 4;
+  using Array = eigen_util::FArray<N>;
+  using Tensor = eigen_util::FTensor<Eigen::Sizes<N, 1>>;
+
+  Array array{{0, 1, 2, 3}};
+  Tensor tensor = eigen_util::reinterpret_as_tensor<Tensor, Array>(array);
+
+  for (int i = 0; i < N; ++i) {
+    EXPECT_EQ(tensor(i, 0), array(i));
+  }
+}
+
+TEST(eigen_util, debug_assert_is_valid_prob_distr_array) {
+  constexpr int N = 4;
+  using Array = eigen_util::FArray<N>;
+
+  Array valid{{0.1, 0.2, 0.3, 0.4}};
+  EXPECT_NO_THROW(eigen_util::debug_assert_is_valid_prob_distr<Array>(valid, 1e-5));
+
+  Array invalid{{0.1, 0.2, 0.3, 0.5}};
+  EXPECT_ANY_THROW(eigen_util::debug_assert_is_valid_prob_distr<Array>(invalid, 1e-5));
+}
+
+TEST(eigen_util, debug_assert_is_valid_prob_distr_tensor) {
+  constexpr int N = 4;
+  using Tensor = eigen_util::FTensor<Eigen::Sizes<N, 1>>;
+
+  Tensor valid;
+  valid.setValues({{0.1}, {0.2}, {0.3}, {0.4}});
+  EXPECT_NO_THROW(eigen_util::debug_assert_is_valid_prob_distr<Tensor>(valid, 1e-5));
+
+  Tensor invalid;
+  invalid.setValues({{0.1}, {0.2}, {0.3}, {0.5}});
+  EXPECT_ANY_THROW(eigen_util::debug_assert_is_valid_prob_distr<Tensor>(invalid, 1e-5));
+}
+
+TEST(eigen_util, sum) {
+  constexpr int M = 2;
+  constexpr int N = 4;
+  using Tensor = eigen_util::FTensor<Eigen::Sizes<M, N>>;
+
+  Tensor tensor;
+  tensor.setValues({{0, 1, 2, 3}, {4, 5, 6, 7}});
+  float expected = 28;
+
+  float result = eigen_util::sum<Tensor>(tensor);
+  EXPECT_EQ(result, expected);
+}
+
+TEST(eigen_util, max) {
+  constexpr int M = 2;
+  constexpr int N = 4;
+  using Tensor = eigen_util::FTensor<Eigen::Sizes<M, N>>;
+
+  Tensor tensor;
+  tensor.setValues({{0, 1, 2, 3}, {4, 5, 6, 7}});
+  float expected = 7;
+
+  float result = eigen_util::max<Tensor>(tensor);
+  EXPECT_EQ(result, expected);
+}
+
+TEST(eigen_util, min) {
+  constexpr int M = 2;
+  constexpr int N = 4;
+  using Tensor = eigen_util::FTensor<Eigen::Sizes<M, N>>;
+
+  Tensor tensor;
+  tensor.setValues({{0, 1, 2, 3}, {4, 5, 6, 7}});
+  float expected = 0;
+
+  float result = eigen_util::min<Tensor>(tensor);
+  EXPECT_EQ(result, expected);
+}
+
+TEST(eigen_util, any) {
+  constexpr int M = 2;
+  constexpr int N = 4;
+  using Tensor = eigen_util::FTensor<Eigen::Sizes<M, N>>;
+
+  Tensor tensor;
+  tensor.setValues({{0, 1, 2, 3}, {4, 5, 6, 7}});
+  bool expected = true;
+
+  bool result = eigen_util::any<Tensor>(tensor);
+  EXPECT_EQ(result, expected);
```

Comment:
You should add a second tensor of all zeros and then verify that `eigen_util::any()` is correct for that one.

It's more idiomatic to use `EXPECT_TRUE` (and `EXPECT_FALSE`) rather than `EXPECT_EQ` here.